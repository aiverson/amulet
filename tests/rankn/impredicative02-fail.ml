(* A different version of lint/shadow_forall.ml, which is caught by the
 * predicativity checker *)

let foo : forall 'a. 'a -> forall 'b. 'b -> 'a = fun x y -> x

let main = (foo foo foo) () ()
