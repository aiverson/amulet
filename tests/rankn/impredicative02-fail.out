impredicative02-fail.ml[6:12 ..6:27]: error
  Illegal instantiation of type variable ak
                  with polymorphic type `{'b : type}. 'b -> 'a`
  Â· Note: doing so would constitute impredicative polymorphism
  
  Arising from use of the expression
